libs/npm/core/src/utils/elastic/generateTerms.ts

```diff
@@ -4,9 +4,15 @@
  * @param values An array of values to search for.
  * @returns An Elasticsearch query.
  */
-export const generateTerms = (field: string, values?: any[]) => {
+export const generateTerms = (field: string, values?: any[], sendEmptyArray: boolean = false) => {
```

Comment:
What is this functionality for?

Reply:
When we send an empty array to a term, this functions suppress the array.
Ex: ids: [] should bring nothing, but since it's suppressed, brings all.
So I added this parameter because ins some cases I really need the array to be empty, because I don't want return for that call.
I added this parameter the keep back compatibility, because some parts of our application, uses this like:
Ex: ids: [] = brings all
When I need to be specific saying:
Ex: When my field is SourceIds, pass (not suppress) an empty array.
I can make it work, and the rest of the application, is not affected.

Reply:
`ids: []` shouldn't bring all if I'm understanding this correctly.  That's telling elastic that we only want content in the array.  If the array is empty, then it shouldn't bring any content.  I think it makes more sense that if it's supposed to bring all, that it should remove `ids` from the filter before sending it.

Reply:
That exactly what this change is from ... but ... changing the default behavior  through the whole application, is not a good idea. A lot of calls does not expect this ...
Maybe in (a near) the future we can revisit all the calls, but a change for everything would blow everything up, and probably transform this in a scope creep.
But for the media types with this PR, id; [] brings empty.